### PR TITLE
UI/javadocs/bugs/features/new code for GUEST APP

### PR DESCRIPTION
**Pull request by mmccoy37**
## Completed:
+ cleaned up code
+ finished ui for FindEmployeeActivity
+ renamed a bunch of variables and a few classes to be more descriptive 
(eg. TabletDirectoryActivity -> MainDirectoryActivity, AsyncGetEmployees -> FullEmployeesByHttpAsyc, PathwayDirectionsActivity -> FindEmployeeActivity, etc) These make it more obvious what they do for anyone unfamiliar with the project.
+ added a bunch of documentation
+ now sorts directory list
+ includes previously unpushed changes which add:
    + fixes for filter
    + backspace button for filter
    + navigation buttons

## Incomplete:
 + code inside CustomListAdapter which allows me to:
    + hide backspace button for filter when no text is entered
    + show image "no employees found" when filter has 0 results

I am using the same context which works for the rest of the class so im not sure why it isnt working. I'll try to finish it if I have time.... would add nice 

## Video (link from slack)
+ https://theplanetexpress.slack.com/files/mmccoy37/F50DD4WMC/2017_04_15_18_05_13.mp4

Edit (17 April):
+ hide card layout until data is retrieved
+ fixes loading message
+ update text in card for color
+ new color utility class for converting our string colors to hex codes form Color.COLOR

### CODE REVIEW

**Comment by mmccoy37** [comment]
tested the code for when it returns a color by faking server response. I fixed the bugs related to it. the colors I setup to display properly in text are red,orange,yellow,green,blue,white,purple.

Color parsing was the only bug and I fixed it. Feel free to test.

File: android-guest-app/app/src/main/java/edu/gatech/ledpathways/ledpathways/TabletDirectoryActivity.java [hunk 1]
```diff
@@ -24,23 +26,25 @@
...
-import static edu.gatech.ledpathways.ledpathways.objects.ServerInfo.SERVER_URL;
 
 /**
  * An example full-screen activity that shows and hides the system UI (i.e.
  * status bar and navigation/system bar) with user interaction.
  */
 public class TabletDirectoryActivity extends AppCompatActivity {
 
-    private String API_URL = SERVER_URL + "/api/employees";
+    private String API_URL = "https://led-pathways.herokuapp.com/api/employees";
```
Comment (mobyvb) [inline android-guest-app/app/src/main/java/edu/gatech/ledpathways/ledpathways/TabletDirectoryActivity.java]: why did you do this?

Comment (mobyvb) [inline android-guest-app/app/src/main/java/edu/gatech/ledpathways/ledpathways/TabletDirectoryActivity.java]: Jk, ignore this comment. It's not in the final diff

**Comment by mobyvb** [comment]
Tested it with colors red, green, and blue for status code 200, and it worked fine.

**Pull request merged**